src/hooks/useFetch.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+
+const BASE_URL = `https://api.disneyapi.dev/character`;
+
+export const useFetch = (page = "1", id) => {
+  const [characters, setCharacters] = useState([]);
+  const [info, setInfo] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [errors, setErrors] = useState(null);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      setLoading(true);
+      try {
+        const resp = await fetch(`${BASE_URL}/${id ? id : ":id"}?page=${page}`);
```

Comment:
The URL template string uses :id which is not a valid placeholder. It should be ${id ? id : ""} to handle the case when id is not provided.